### ISSUE TITLE
Deprecation warnings

### ISSUE THREAD

**Issue by marianodonal** [ishanvyas22/asset-mix#34]
**Describe the bug**
Hi, i'm new with react and laravel mix.
I'm trying to create an app when I get two deprecation warnings:

`Router::plugin()` is deprecated, use the non-static method `RouterBuilder::plugin()` instead. - /Applications/MAMP/htdocs/cakereact/vendor/ishanvyas22/asset-mix/config/routes.php, line: 11
 You can disable all deprecation warnings by setting `Error.errorLevel` to `E_ALL & ~E_USER_DEPRECATED`, or add `vendor/ishanvyas22/asset-mix/config/routes.php` to  `Error.ignoredDeprecationPaths` in your `config/app.php` to mute deprecations from only this file. [CORE/src/Core/functions.php, line 322]

`Router::scope()` is deprecated, use the non-static method `RouterBuilder::scope()` instead. - /Applications/MAMP/htdocs/cakereact/vendor/cakephp/cakephp/src/Routing/Router.php, line: 912
 You can disable all deprecation warnings by setting `Error.errorLevel` to `E_ALL & ~E_USER_DEPRECATED`, or add `vendor/cakephp/cakephp/src/Routing/Router.php` to  `Error.ignoredDeprecationPaths` in your `config/app.php` to mute deprecations from only this file. [CORE/src/Core/functions.php, line 322]

**Comment by ishanvyas22** [comment]
What is exact CakePHP version you are using?

**Comment by marianodonal** [comment]
CakePHP 4.3.0

**Comment by marianodonal** [comment]
I've fixed it by updating routes.php file in plugin config folder.

```
use Cake\Routing\RouteBuilder;
use Cake\Routing\Route\DashedRoute;

return static function (RouteBuilder $routes) {
    $routes->plugin('AssetMix', ['path' => '/asset-mix'], function (RouteBuilder $routes) {
        $routes->fallbacks(DashedRoute::class);
    });
};
```

I don't know how to perform a pull request, maybe I can collaborate. Thanks

**Comment by Andream98** [comment]
I've also changed the routes file because of the deprecation warnings I get when I start the CakePHP server:

```
<?php

    use Cake\Routing\Route\DashedRoute;
    use Cake\Routing\RouteBuilder;

    /** @var RouteBuilder $routes */
    $routes->plugin(
        'Inertia',
        ['path' => '/inertia'],
        function (RouteBuilder $routes) {
            $routes->fallbacks(DashedRoute::class);
        }
    );
```

@ishanvyas22 let me know which version seems more appropriate to you so that I can create a PR.

PS: I will be opening the same PR on [cakephp-inertiajs](https://github.com/ishanvyas22/cakephp-inertiajs) too

**Comment by ishanvyas22** [comment]
@Andream98 v4.3.0 seems good as [it is deprecated](https://github.com/cakephp/cakephp/blob/4.x/src/Routing/Router.php#L914) in that version only.

You can create a PR with the change and updating CakePHP version to 4.3 in `composer.json` file.

**Comment by ishanvyas22** [comment]
Released in [1.4.0](https://github.com/ishanvyas22/asset-mix/releases/tag/1.4.0)
